langchain/cache.py

```diff
@@ -26,6 +26,39 @@ def _hash(_input: str) -> str:
     return hashlib.md5(_input.encode()).hexdigest()
 
 
+def _dump_generations_to_json(generations: RETURN_VAL_TYPE) -> str:
+    """Dump generations to json.
+
+    Args:
+        generations (RETURN_VAL_TYPE): A list of language model generations.
+
+    Returns:
+        str: Json representing a list of generations.
+    """
+    return json.dumps([generation.dict() for generation in generations])
+
+
+def _load_generations_from_json(generations_json: str) -> RETURN_VAL_TYPE:
+    """Load generations from json.
+
+    Args:
+        generations_json (str): A string of json representing a list of generations.
+
+    Raises:
+        ValueError: Could not decode json string to list of generations.
+
+    Returns:
+        RETURN_VAL_TYPE: A list of generations.
+    """
+    try:
+        results = json.loads(generations_json)
+        return [Generation(**generation_dict) for generation_dict in results]
+    except json.JSONDecodeError:
+        raise ValueError(
+            f"Could not decode json to list of generations: {generations_json}"
+        )
```

Comment:
I extracted these to functions. I was going to refactor the GPTCache plugin to use these, but found I cannot run their integration tests. They fail on my end and are possibly broken.